scripts/create-python-mirror.py

```diff
@@ -89,6 +89,16 @@ def collect_metadata_from_git_history() -> List[Dict]:
     return metadata
 
 
+def check_arch(entry, arch):
+    """Checks whether arch entry in metadata matches the provided filter."""
+    match entry:
+        case {"family": family}:
+            return family == arch
+        case str:
```

Comment:
Can't use because of other issue with supporting 3.8 and 3.9 - but for future reference keep in mind that this does not test that the input to `match` is a string. It simply assigns str as a variable to the input - which should be flagged as reassigning over a default name.

To test that something is a string do:

```python
match x:
    case variable if isinstance(variable, str):
        print(f"input was a string: {variable}")
    case _:
        print(f"input was NOT a string: {x}"
```